stl/src/filesystem.cpp

```diff
@@ -9,6 +9,7 @@
 // Do not include or define anything else here.
 // In particular, basic_string must not be included here.
 
+#include <algorithm>

```

Comment:
`<algorithm>` is an extremely large header, and injecting it into the import lib is concerning (although it doesn't necessarily pose the ODR concerns that `<string>` does).

It looks like it's only needed for `find()` and `find_if()`. If you extract the core implementation of `find_if()` to `<xfilesystem_abi.h>` (see other comment), then you could use that with a lambda to replace `find()`, or you could also extract the core implementation of `find()` (no checking/vectorization/etc).